src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -150,7 +153,39 @@ private void OnGetSuggestionsCommandExecute()
                 _suggestionsDropdownTelemetry.DropdownButtonClickTimes += 1;
             }
 
-            _getSuggestionsTask = _smartRenameSession.GetSuggestionsAsync(_cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
+            var allRenameLocations = BaseViewModel.Session.AllRenameLocationsTask.Join();
+            ImmutableDictionary<string, string[]> context = default;
+            if (allRenameLocations.Locations.Count > 0)
+            {
+                var references = new List<string>();
+                foreach (var renameLocation in allRenameLocations.Locations)
+                {
+                    var syntaxTree = renameLocation.Document.GetSyntaxTreeSynchronously(_cancellationTokenSource.Token);
+                    var text = syntaxTree.GetText(_cancellationTokenSource.Token);
+                    var lineSpan = syntaxTree.GetLineSpan(renameLocation.TextSpan, _cancellationTokenSource.Token);
+
+                    var startLine = lineSpan.StartLinePosition.Line;
+                    var endLine = lineSpan.EndLinePosition.Line;
+
+                    var documentContent = string.Join(
+                        Environment.NewLine,
+                        text.Lines.Skip(startLine).Take(endLine - startLine + 1).Select(l => l.ToString()));
+                    references.Add(documentContent);
+                }
+                var contextBuilder = ImmutableDictionary.CreateBuilder<string, string[]>();
+                contextBuilder.Add("Reference", references.ToArray());
+                context = contextBuilder.ToImmutableDictionary();
```

Comment:
this seems to be a dictionary with one elemnt in it.  Can just do `ImmutableDictionary.Empty.Add`

Reply:
this is now refactored